src/app/components/normal-component/welcome/welcome.component.spec.ts

```diff
@@ -30,6 +36,25 @@ describe('WelcomeComponent', () => {
 
         expectValidRouting(router, ['/play', 'Awale'], OnlineGameCreationComponent);
     }));
+    it('should not redirect to online game creation when clicking on the corresponding button while in a game', fakeAsync(async() => {
+        // Given a welcome component
+        // where ConnectedUserService tells us user cannot join a game
+        const connectedUserService: ConnectedUserService = TestBed.inject(ConnectedUserService);
+        const error: string = `j'ai dit non!`;
+        spyOn(connectedUserService, 'canUserCreate').and.returnValue(MGPValidation.failure(error));
+        const router: Router = TestBed.inject(Router);
+        spyOn(router, 'navigate').and.callThrough();
+
+        // When clicking on the online-button of one game
+        const component: WelcomeComponent = testUtils.getComponent();
+        spyOn(component.messageDisplayer, 'criticalMessage').and.callThrough();
+        await testUtils.clickElement('#playOnline_Awale');
+        tick(3000);
+
+        // Then the component should not have changed page and toast the reason
```

Comment:
and *should* toast the reason (multiple times)

Reply:
two times ?